GitHubTeams.ps1

```diff
@@ -153,8 +161,39 @@ filter Get-GitHubTeam
         'NoStatus' = (Resolve-ParameterWithDefaultConfigurationValue -Name NoStatus -ConfigValueName DefaultNoStatus)
     }
 
-    return (Invoke-GHRestMethodMultipleResult @params |
-        Add-GitHubTeamAdditionalProperties)
+    $result = Invoke-GHRestMethodMultipleResult @params | Add-GitHubTeamAdditionalProperties
+
+    if ($PSBoundParameters.ContainsKey('TeamName'))
+    {
+        $team = $result | Where-Object -Property name -eq $TeamName
+
+        if ($null -eq $team)
+        {
+            $message = "Team '$TeamName' not found"
+            Write-Log -Message $message -Level Error
+            throw $message
+        }
+        else
+        {
+            $uriFragment = "/teams/$($team.id)"
+            $description = "Getting team $($team.id)"
+
+            $params = @{
+                UriFragment = $uriFragment
+                Description =  $description
+                Method = 'Get'
+                AccessToken = $AccessToken
+                TelemetryEventName = $MyInvocation.MyCommand.Name
+                TelemetryProperties = $telemetryProperties
+                NoStatus = (Resolve-ParameterWithDefaultConfigurationValue `
+                    -Name NoStatus -ConfigValueName DefaultNoStatus)
+            }
+
+            $result = Invoke-GHRestMethod @params | Add-GitHubTeamAdditionalProperties
```

Comment:
We had [talked about](https://github.com/microsoft/PowerShellForGitHub/pull/222#discussion_r446565393) this being saved as a `GitHub.TeamDetail` object.  You can just use the current one and override TypeName. You should probably update `Add-GitHubTeamAdditionalProperties` to add the additional properties to $item.organization if it's non-null.